modules/calcite/src/test/java/org/apache/ignite/internal/processors/query/calcite/CalciteQueryProcessorTest.java

```diff
@@ -852,6 +876,112 @@ public void testExceptBigBatch() throws Exception {
         assertEquals(128, F.size(rows, r -> r.get(0).equals(2)));
         assertEquals(1920, F.size(rows, r -> r.get(0).equals(3)));
         assertEquals(32768, F.size(rows, r -> r.get(0).equals(4)));
+
+        rows = sql("SELECT _val FROM \"cache1Replicated\".table1_repl INTERSECT SELECT _val FROM \"cache2Replicated\"" +
+            ".table2_repl");
+
+        assertEquals(2, rows.size());
+        assertEquals(1, F.size(rows, r -> r.get(0).equals(1)));
+        assertEquals(1, F.size(rows, r -> r.get(0).equals(3)));
+
+        rows = sql("SELECT _val FROM \"cache1Replicated\".table1_repl INTERSECT ALL SELECT _val FROM \"cache2Replicated\"" +
+            ".table2_repl");
+
+        assertEquals(136, rows.size());
+        assertEquals(8, F.size(rows, r -> r.get(0).equals(1)));
+        assertEquals(128, F.size(rows, r -> r.get(0).equals(3)));
+    }
+
+    /** */
+    @Test
+    public void testIntersect() throws Exception {
```

Comment:
I would prefer not to extend a CalciteQueryProcessorTest since it's already a quite messy and most of the tests should be revised. Let's move those newly created tests to new class with name SetOpIntegrationTest